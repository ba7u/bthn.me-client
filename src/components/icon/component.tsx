import { Icon } from '@iconify/react';
import classNames from 'classnames';
import { IconComponentPropsInterface } from './component.props.interface';

export const IconComponent: React.FunctionComponent<IconComponentPropsInterface> = ({ className, icon }) => (
  <Icon className={classNames(className, 'w-6', 'h-6')} icon={icon} />
);
