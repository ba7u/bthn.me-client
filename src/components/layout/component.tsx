import { motion } from 'framer-motion';

const variants = {
  hidden: { opacity: 0, x: 0, y: 100 },
  enter: { opacity: 1, x: 0, y: 0 },
  exit: { opacity: 0, x: 0, y: 100 },
};

export const LayoutComponent: React.FunctionComponent = ({ children }) => {
  return (
    <div>
      <motion.main
        variants={variants}
        initial='hidden'
        animate='enter'
        exit='exit'
        transition={{ type: 'linear' }}
        className='
          flex flex-col items-start w-full pt-10
          px-8 sm:px-16 md:px-36 lg:px-52 xl:px-80 2xl:px-96
          pt-24 h-full
        '
      >
        {children}
      </motion.main>
    </div>
  );
};
