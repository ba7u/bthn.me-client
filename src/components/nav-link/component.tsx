import Link from 'next/link';
import { NavLinkComponentPropsInterface } from './component.props.interface';

export const NavLinkComponent: React.FunctionComponent<NavLinkComponentPropsInterface> = ({
  isActive,
  children,
  activeComponent,
  ...props
}) => (
  <Link {...props} scroll={false}>
    <a className='mr-6 sm:mr-8 flex flex-col relative'>
      {children}
      {isActive && activeComponent}
    </a>
  </Link>
);
