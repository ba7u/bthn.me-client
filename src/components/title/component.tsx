import Link from 'next/link';

export const TitleComponent: React.FunctionComponent = () => (
  <Link href='/'>
    <a>
      <span>Batuhan Toparlak</span>
    </a>
  </Link>
);
