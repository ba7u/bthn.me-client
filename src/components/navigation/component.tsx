import { AnimateSharedLayout } from 'framer-motion';
import { NavLinkComponent } from 'components/nav-link/component';
import { linkIsActive } from 'helpers/link/is-active';
import { useTranslation } from 'helpers/translation/hook';
import { NavigationComponentPropsInterface } from './component.props.interface';
import { NavigationUnderlineComponent } from './underline.component';

export const NavigationComponent: React.FunctionComponent<NavigationComponentPropsInterface> = ({
  paths,
  activePath,
}) => {
  const { t } = useTranslation('common');

  return (
    <AnimateSharedLayout>
      <nav className='flex'>
        {paths.map(({ textNode, path }) => (
          <NavLinkComponent
            key={textNode}
            href={path}
            isActive={linkIsActive(path, activePath)}
            activeComponent={<NavigationUnderlineComponent />}
          >
            {t(textNode)}
          </NavLinkComponent>
        ))}
      </nav>
    </AnimateSharedLayout>
  );
};
