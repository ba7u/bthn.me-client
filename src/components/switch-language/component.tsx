import { SwitchLanguageComponentPropsInterface } from './component.props.interface';
import { IconComponent } from 'components/icon/component';
import { localeGetAlternative } from 'helpers/locale/get-alternative';
import { NavLinkComponent } from 'components/nav-link/component';

export const SwitchLanguageComponent: React.FunctionComponent<SwitchLanguageComponentPropsInterface> = ({ router }) => {
  return (
    <NavLinkComponent href={router.asPath} locale={localeGetAlternative(router.locale, router.locales)}>
      {router.locale === router.defaultLocale ? (
        <IconComponent icon='emojione-v1:flag-for-turkey' />
      ) : (
        <IconComponent icon='emojione-v1:flag-for-united-kingdom' />
      )}
    </NavLinkComponent>
  );
};
