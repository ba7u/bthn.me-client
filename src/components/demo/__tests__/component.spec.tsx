/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';
import { mockMiscAddTranslation } from 'mocks/misc/add-translation.mock';

import { DemoComponent } from '../component';

describe('DemoComponent', () => {
  beforeAll(() => {
    mockMiscAddTranslation('test', 'lorem');
  });
  it('renders without throwing any errors', () => {
    const { container } = render(<DemoComponent />);

    expect(container).toMatchSnapshot();
  });
});
