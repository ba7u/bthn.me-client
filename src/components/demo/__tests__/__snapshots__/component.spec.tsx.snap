// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoComponent renders without throwing any errors 1`] = `
<div>
  <div>
    <span>
      DemoComponent - Desktop
    </span>
    <span>
      lorem
    </span>
  </div>
</div>
`;
